runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/DispatchAgent.java

```diff
@@ -867,7 +902,10 @@ private void onSystemSignal(
         switch (signalId)
         {
         case SIGNAL_TASK_QUEUED:
-            taskQueue.poll().run();
+            if (!taskQueue.isEmpty())
+            {
+                taskQueue.poll().run();
+            }
```

Comment:
What triggered the need for this change? Feels like we might be working around a bug.

Reply:
Bug found, and resolved. The WatcherTask didn't close properly as ForkJoinTask.cancel is not working as expected.